blocks/faq/src/app.tsx

```diff
@@ -0,0 +1,469 @@
+    shouldDisplayQuestionNumbers ?? true,
+  );
+  const [displayToggles, setDisplayToggles] = useState(
+    shouldDisplayQuestionToggles ?? true,
+  );
+  const [questions, setQuestions] = useState<
+    {
+      id: string;
+      properties: FrequentlyAskedQuestionProperties;
+      animatingOut?: boolean;
+    }[]
+  >(
+    questionEntities.map((questionEntity) => ({
+      id: questionEntity.metadata.recordId.entityId,
+      properties: questionEntity.properties,
+    })),
+  );
+
+  const updateField = async (value: string | boolean, field: RootEntityKey) => {
+    await graphModule.updateEntity({
+      data: {
+        entityId,
+        entityTypeId,
+        properties: {
+          ...properties,
+          [field]: value,
+        },
+      },
+    });
+  };
+
+  const createFrequentlyAskedQuestionEntity = useCallback(async () => {
+    if (readonly) {
+      return;
+    }
+
+    const createEntityResponse = await graphModule.createEntity({
+      data: {
+        entityTypeId: frequentlyAskedQuestionType,
+        properties: {},
+      },
+    });
+
+    const createdEntityId =
+      createEntityResponse.data?.metadata.recordId.entityId;
+
+    if (createdEntityId) {
+      await graphModule.createEntity({
+        data: {
+          entityTypeId: hasFrequentlyAskedQuestion,
+          properties: {},
+          linkData: {
+            leftEntityId: entityId,
+            rightEntityId: createdEntityId,
+            leftToRightOrder:
+              (questionLinkedEntities[questionLinkedEntities.length - 1]
+                ?.linkEntity.linkData?.leftToRightOrder ?? 0) + 1,
+          },
+        },
+      });
+    }
+  }, [graphModule, questionLinkedEntities, entityId, readonly]);
+
+  const addQuestion = useCallback(async () => {
+    setQuestions([...questions, { id: uuid(), properties: {} }]);
+    await createFrequentlyAskedQuestionEntity();
+  }, [questions, createFrequentlyAskedQuestionEntity]);
+
+  const updateQuestionField = async (
+    index: number,
+    value: string,
+    field: QuestionOrAnswer,
+  ) => {
+    const questionEntity = questionEntities[index];
+
+    if (!questionEntity) {
+      return;
+    }
+
+    await graphModule.updateEntity({
+      data: {
+        entityId: questionEntity.metadata.recordId.entityId,
+        entityTypeId: frequentlyAskedQuestionType,
+        properties: {
+          ...questionEntity.properties,
+          [field]: value,
+        },
+      },
+    });
+  };
+
+  const removeQuestion = (index: number) => {
+    setQuestions(
+      questions.map((question, questionIndex) => {
+        if (index === questionIndex) {
+          return { ...question, animatingOut: true };
+        }
+        return question;
+      }),
+    );
+  };
+
+  useEffect(() => {
+    if (!questionEntities.length) {
+      void addQuestion();
+    }
+    // We only want to run this once when the block is initiated
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Do we also want to run this on `readonly` mode?. 
This is probably an edge case, but what should happen if we render an empty question list in `readonly` mode?